feat: add temporal metrics and tracing docs for Signoz (#1208)
SigNoz/signoz.io: docs/userguide/hostmetrics.md

---
id: hostmetrics
title: Hostmetrics Dashboard
description: View your VM's hostmetrics in SigNoz Cloud and Self-host

---

import Tabs from "@theme/Tabs";
import TabItem from "@theme/TabItem";

<figure data-zoomable align='center'>
    <img src="/img/hostmetrics-dashboard-overview.webp" alt="Overview of the Hostmetrics Dashboard in SigNoz which can be used to see metrics like CPU Usage, Memory Usage, Disk I/O, Filesystem Usage, Network I/O"/>
    <figcaption><i>Overview of Hostmetrics Dashboard in SigNoz</i></figcaption>
</figure>
<br></br>

## Overview

The **Host Metrics dashboard** consists of charts for monitoring instance metrics and is designed as a generic dashboard with a hostname variable. You can access detailed information on common host metrics such as :

- CPU Usage
- Memory Usage
- Disk I/O
- Filesystem Usage
- Network I/O

> A full list of supported hostmetrics can be found <a href = "https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/hostmetricsreceiver/README.md#getting-started" rel="noopener noreferrer nofollow" target="_blank" >here</a>

Reviewer: ```suggestion
A full list of supported hostmetrics can be found <a href = "https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/hostmetricsreceiver/README.md#getting-started" rel="noopener noreferrer nofollow" target="_blank" >here</a>.
```

## Setup

> You can choose from the two options below

Reviewer: ```suggestion
You can choose from the two options below.
```

<Tabs>
<TabItem value="SigNoz Cloud" label="SigNoz Cloud" default>

> A receiver serves as the entry point for data into the OpenTelemetry Collector. The Host Metrics receiver generates metrics about the host system scraped from various sources, like your VM, and is intended to be used when the collector is deployed as an agent.

Reviewer: ```suggestion
A receiver serves as the entry point for data into the OpenTelemetry Collector. The Host Metrics receiver generates metrics about the host system scraped from various sources, like your VM, and is intended to be used when the collector is deployed as an agent.
```

@srikanthccv can you review the paragraph here? Can we say receiver is an entry point or is it just a config. 

@Calm-Rock I am not sure about the second line - `The Host Metrics receiver generates metrics about the host system scraped from various sources, like your VM`. If the receiver generates the metrics, what does it scrap? I think the receiver just scrapes metrics, there is no generation involved. @srikanthccv can confirm.

Reviewer 2: Yes, you are correct. It doesn't generate any metric on its own. The scraper receiver is a kind of receiver that periodically tries to fetch the metrics from some source (which generates the metrics). The host metrics receiver scrapes the metrics from the host machine.

Reviewer 2: Your interpretation of what metric generation seems to be different. The metrics are generated by kernel modules and are written to particular files. The host metrics simply read them and create OTLP-compatible data. This applies to other scrapers as well. For instance, A Postgres DB server is the actual process that generates the metrics and saves them to some monitoring DB which is read by the collector and creates OTLP data.

Reviewer 2: In the context of host metrics, the source here is not a cross-platform library but a file that contains the info example `/sys/block/sda/size`, `/proc/...` etc. Whatever the cross-platform thing you mention is an library that makes it easier to read these files on different operating systems.

Reviewer 2: The source for different scrapers is different, for some, it's some files, and for others, it's some table in DB. And third-party library only helps read such data.

Reviewer 2: @ankit01-oss saying host metrics receiver generates metrics is as accurate as saying file log receiver generates logs. Neither of them generates anything on their own. They simply collect the data generated by other processes and format them in an OTLP-compatible format.

**Step 2** : Download/Copy the hostmetrics json file

Hostmetric json file will help you create a pre-built dashboard about all the <a href = "https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/hostmetricsreceiver/README.md#getting-started" rel="noopener noreferrer nofollow" target="_blank" >categories</a> of host metrics provided by hostmetrics receiver.

Use this <a href = "https://github.com/SigNoz/dashboards/blob/main/hostmetrics/hostmetrics-with-variable.json" rel="noopener noreferrer nofollow" target="_blank" >GitHub Link</a> to download/copy the `hostmetrics-with-variable.json` file. The structure of the json file is shown below.

<figure data-zoomable align='center'>
    <img src="/img/hostmetrics-with-variable-json.webp" alt="The structure of json file for the  Hostmetrics Dashboard for SigNoz UI"/>
    <figcaption><i>Structure of hostmetrics-with-variable.json</i></figcaption>
</figure>
<br></br>

> **Step 3** : Import hostmetrics json file to SigNoz Cloud

Reviewer 3: can you also update other usage of `json` with `JSON`?

<figure data-zoomable align='center'>
    <img src="/img/hostmetrics-new-dashboard.webp" alt="How to create a New Hostmetrics Dashboard in SigNoz UI"/>
    <figcaption><i>Create a New Dashboard by importing JSON</i></figcaption>
</figure>
<br></br>

Upload/paste your `hostmetrics-with-variable.json` data and click on **Load JSON** button

<figure data-zoomable align='center'>
    <img src="/img/hostmetrics-load-json.webp" alt="Load the Hostmetrics Dashboard JSON file in SigNoz UI"/>
    <figcaption><i>Load the Hostmetrics JSON file</i></figcaption>
</figure>
<br></br>

After loading the JSON, you can find your hostname in the  **<i>$hostname</i>** dropdown menu. Select it, and you'll access the Hostmetrics Dashboard for the VM where you've configured the OTel collector.

<figure data-zoomable align='center'>
    <img src="/img/hostmetrics-select-hostname.webp" alt="Select hostname from Hostmetrics Dashboard in SigNoz UI"/>
    <figcaption><i> Select hostname from the dropdown</i></figcaption>
</figure>
<br></br>

</TabItem>

<TabItem value="Self-Host" label="Self-Host">

> A receiver serves as the entry point for data into the OpenTelemetry Collector. The Host Metrics receiver generates metrics about the host system scraped from various sources, like your VM, and is intended to be used when the collector is deployed as an agent.

Reviewer 3: same as above

> There two ways in which the Self-Host version of SigNoz can be configured :

Reviewer: ```suggestion
There two ways in which the Self-Host version of SigNoz can be configured:

- SigNoz is installed on the same VM as your application
- SigNoz is installed on a different VM
```

#### On Same VM

> When SigNoz is running on the same VM as the Host, it automatically starts collecting the Hostmetrics data because the hostmetrics receiver is enabled by default in the OTel collector that comes bundeled with SigNoz.

Reviewer 3: ```suggestion
When SigNoz is running on the same VM as the Host, it automatically starts collecting
the Hostmetrics data because the hostmetrics receiver is enabled by default in the
OTel collector configuration that comes bundled with SigNoz.
```

@Calm-Rock Breaking long lines helps in review. One line should not exceed 100 chars, while 70-80 chars being ideal per line.

<figure data-zoomable align='center'>
    <img src="/img/hostmetrics-new-dashboard.webp" alt="How to create a New Hostmetrics Dashboard in SigNoz UI"/>
    <figcaption><i>Create a New Dashboard by importing JSON</i></figcaption>
</figure>
<br></br>

Upload/paste your `hostmetrics-with-variable.json` data and click on **Load JSON** button

<figure data-zoomable align='center'>
    <img src="/img/hostmetrics-load-json.webp" alt="Load the Hostmetrics Dashboard JSON file in SigNoz UI"/>
    <figcaption><i>Load the Hostmetrics JSON file</i></figcaption>
</figure>
<br></br>

After loading the JSON, you can find your hostname in the **<i>$hostname</i>** dropdown menu. Select it, and you'll access the Hostmetrics Dashboard for the VM where you've configured the OTel collector.

<figure data-zoomable align='center'>
    <img src="/img/hostmetrics-select-hostname.webp" alt="Select hostname from Hostmetrics Dashboard in SigNoz UI"/>
    <figcaption><i> Select hostname from the dropdown</i></figcaption>
</figure>
<br></br>

</TabItem>
</Tabs>

## Dashboard Configurations

> Click on **Configure** button at top right corner to change the name, add/remove tags, edit description, and add more variables to your dashboard. These things can be edited before uploading the JSON file or from the dashboard itself.

Reviewer: ```suggestion
These things can be edited before uploading the JSON file or from the dashboard itself.
```

Not sure if this line is required. I think if we are enabling an easier process from our dashboard, explaining that should be enough.

<figure data-zoomable align='center'>
    <img src="/img/hostmetrics-configure-general-section.webp" alt="Select hostname from Hostmetrics Dashboard in SigNoz UI"/>
    <figcaption><i> Hostmetrics General Settings</i></figcaption>
</figure>
<br></br>

<figure data-zoomable align='center'>
    <img src="/img/hostmetrics-configure-variables-section.webp" alt="Select hostname from Hostmetrics Dashboard in SigNoz UI"/>
    <figcaption><i> Hostmetrics Variables Settings</i></figcaption>
</figure>
<br></br>

> To explore more dashboard functionalities, check out the  <a href = "https://signoz.io/docs/userguide/manage-dashboards-and-panels/" rel="noopener noreferrer nofollow" target="_blank" >Dashboard Section</a>

Reviewer: ```suggestion
To explore more dashboard functionalities, check out the  <a href = "https://signoz.io/docs/userguide/manage-dashboards-and-panels/" rel="noopener noreferrer nofollow" target="_blank" >Dashboard Section</a>.
```
